library/src/main/java/com/mux/video/upload/internal/UploadMetrics.kt

```diff
@@ -49,10 +54,32 @@ internal class UploadMetrics private constructor() {
       regionCode = Locale.getDefault().country
     ).toJson()
 
-    // For this case, redirect-following is desired
     val httpClient = MuxUploadSdk.httpClient().newBuilder()
+      // The SDK's http client is configured for uploading. We want the tighter default timeouts
+      .callTimeout(0, TimeUnit.SECONDS)
+      .writeTimeout(10, TimeUnit.SECONDS)
+      // Here, redirect-following is desired. Currently we do a special one, but keep defaults also
       .followRedirects(true)
       .followSslRedirects(true)
+      // We need to do a non-compliant redirect: 302 but preserving method and body
```

Comment:
Hypothetically, if we were to get a 307 instead of a 302 this workaround be necessary?

Reply:
Hypothetically, I think it would work yeah 🤔

Reply:
Cool was just wondering because exact form of redirect we end up using [came up](https://github.com/muxinc/swift-upload-sdk/pull/3#discussion_r1147853746) before. We cannot change the returned status code to 307 for now though.